ios/ReactCBLite.m

```diff
@@ -59,6 +61,91 @@ @implementation ReactCBLite
+    CBLDatabase *database = [manager databaseNamed:dbName error:&error];
+    if (error) {
+        NSLog(@"Error opening database %@. %@", dbName, error);
+        reject(@"Error opening database", [error localizedFailureReason], error);
+        return;
+    }
+    CBLReplication *pushReplication = [database createPushReplication:[NSURL URLWithString:urlString]];
+    pushReplication.headers = headers;
+    pushReplication.continuous = YES;
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(replicationChanged:)
+                                                 name:kCBLReplicationChangeNotification
+                                               object:pushReplication];
+    [pushReplication start];
+    resolve(@[]);
+}
+
+- (void) replicationChanged:(NSNotification *)notification {
+    CBLReplication *replication = notification.object;
+    switch (replication.status) {
+        case kCBLReplicationStopped: {
+            NSDictionary *userInfo = @{
+                                       @"status": @"kCBLReplicationStopped",
+                                       @"pullReplication": @(replication.pull)
+                                       };
+            [[NSNotificationCenter defaultCenter] postNotificationName:kReactCBLiteReplicationChangeNotification
+                                                                object:userInfo];
+            break;
+        }
+        case kCBLReplicationIdle:
+        {
+            NSDictionary *userInfo = @{
+                                       @"status": @"kCBLReplicationIdle",
+                                       @"pullReplication": @(replication.pull)
+                                       };
+            [[NSNotificationCenter defaultCenter] postNotificationName:kReactCBLiteReplicationChangeNotification
+                                                                object:userInfo];
+            break;
+        }
+        case kCBLReplicationActive: {
+            NSDictionary *userInfo = @{@"status": @"kCBLReplicationActive",
+                                       @"pullReplication": @(replication.pull)
+                                       };
+            [[NSNotificationCenter defaultCenter] postNotificationName:kReactCBLiteReplicationChangeNotification
+                                                                object:userInfo];
+            break;
+        }
+        case kCBLReplicationOffline: {
+            NSDictionary *userInfo = @{@"status": @"kCBLReplicationOffline",
+                                       @"pullReplication": @(replication.pull)
+                                       };
+            [[NSNotificationCenter defaultCenter] postNotificationName:kReactCBLiteReplicationChangeNotification
+                                                                object:userInfo];
+            break;
+        }
+        default:
+            break;
+    }
+}
+
```

Comment:
there is a lot of replication (pun intended) in this I would pull the definition of the Notification outside of the switch statement so that all you do inside the switch is set the value of the status.
If the status is set inside the switch then you can send the appropriate notification.  This will make the overall code cleaner and clearer I think (and reduce that line count by a chunk